test/Common/ServerFixture.cs

```diff
@@ -44,32 +42,22 @@ private void StartServer()
                 .UseContentRoot(Directory.GetCurrentDirectory())
                 .Build();
 
-            var t = Task.Run(() => _host.Start());
             _logger.LogInformation("Starting test server...");
-            _lifetime = _host.Services.GetRequiredService<IApplicationLifetime>();
-            if (!_lifetime.ApplicationStarted.WaitHandle.WaitOne(TimeSpan.FromSeconds(5)))
+            using (var cts = new CancellationTokenSource())
             {
-                // t probably faulted
-                if (t.IsFaulted)
-                {
-                    throw t.Exception.InnerException;
-                }
-                throw new TimeoutException("Timed out waiting for application to start.");
+                cts.CancelAfter(TimeSpan.FromSeconds(30));
+                _host.StartAsync(cts.Token).GetAwaiter().GetResult();
```

Comment:
Wouldn't throw when the token is cancelled? Do we need to catch this exception (or any other exception actually) log and swallow?

Reply:
We do something similar to what we did before
```
try
{
    _host.StartAsync(cts.Token).GetAwaiter().GetResult();
}
catch (TaskCanceledException)
{
    throw new TimeoutException("Timed out waiting for application to start.");
}
```

Reply:
I think it would be useful because it immediately tells what happened. With TaskCanceledException you would have to go to the code to understand what really happened.

Reply:
Lets just remove the whole thing. I don't understand why our Startup would ever hang. If you have a good reason, let me know because none of the other repositories do this AFAIK.

Reply:
I just talked with @BrennanConroy and I think we need to take a second look at this because we could not find the reason why the original approach would not work and why the failures would be fixed by this change.

Reply:
We should simplify the code regardless. None of the existing code is required.

Reply:
Copied from MVC :P